### ISSUE TITLE
[Modal] Multiple modals always renders the last one in embedded apps

### ISSUE THREAD

**Issue by mbaumbach** [Shopify/polaris-react#2009]
## Issue summary

This is a specific issue with the `@shopify/app-bridge-react` components, but I'm not sure where to submit bugs for that. Feel free to point me in a better direction.

When using the `Modal` component from the app bridge react components in an embedded app, if you have multiple modals rendering it will always display the last modal in the component hierarchy even when specifying the `open` prop for any others.

## Expected behavior

When specifying the `open` prop to `true` on a `Modal`, that modal and its content should be displayed. Any other `Modal` in the component hierarchy with the `open` prop set to `false` should not be displayed.

## Actual behavior

No matter what `Modal` has the `open` prop set to `true`, the last modal rendered in the entire component hierarchy is the one that is displayed. This is true for components that are deeply nested too. You may change a prop on a `Modal` in a component based on local state and some other `Modal` in a disparate part of the component hierarchy will display if it's rendered later.

## Steps to reproduce the problem

1. Create a component that contains two `Modal` components and the `open` prop on each is controlled by a different local boolean state variable.
1. Have some action that triggers the first referenced `Modal` to have its `open` prop set to `true`.
1. The second `Modal` will display after the trigger.

## Reduced test case

```
import React, { Component } from 'react';
import {
  AppProvider,
  Page,
  Layout,
  Button
} from '@shopify/polaris';
import {
  Provider,
  Modal
} from '@shopify/app-bridge-react';

const API_KEY = 'CHANGE_ME_1';
const SHOP_ORIGIN = 'CHANGE_ME_2';

export default class App extends Component {
  state = {
    showModal1: false,
    showModal2: false
  }
  
  render() {
    const { showModal1, showModal2 } = this.state;
    const config = {
      apiKey: API_KEY,
      shopOrigin: SHOP_ORIGIN,
      forceRedirect: true 
    };
    return (
      <AppProvider>
        <Provider config={config}>
          <Page>
            <Layout sectioned>
              <Button onClick={() => this.setState({ showModal1: true })} fullWidth>Display modal</Button>
            </Layout>
            <Modal
              title="Modal 1"
              message="Modal 1"
              onClose={() => this.setState({ showModal1: false })}
              open={showModal1} />
            <Modal
              title="Modal 2"
              message="Modal 2"
              onClose={() => this.setState({ showModal2: false })}
              open={showModal2} />
          </Page>
        </Provider>
      </AppProvider>
    );
  }
}
```

Clicking the `Button` sets the `showModal1` state to `true` and should display "Modal 1" but "Modal 2" shows up instead. Conditionally rendering the modals does open the correct `Modal`, but closing them programmatically with the `open` prop isn't really possible unless you do it across two separate renders.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 3.21.1 (App Bridge React: 1.6.7)
- Browser: Google Chrome 76.0.3809.132 (Official Build) (64-bit)
- Device: MacBook Pro
- Operating System: macOS Mojave 10.14.6

**Comment by vividviolet** [comment]
That's a bug in the component - it should call to close the modal when unmounting. This one is an easier fix - I'll update this issue when we make some progress on it.

**Comment by joshlbaker** [comment]
Any update on this?

**Comment by vividviolet** [comment]
I'm looking into a fix today

**Comment by joshlbaker** [comment]
Thanks for fixing this @vividviolet !!

**Comment by vividviolet** [comment]
This issue is now fixed in @shopify/app-bridge v1.11.0: https://www.npmjs.com/package/@shopify/app-bridge/v/1.11.0

@mbaumbach You should now be able to render multiple modals at the same time or conditionally render modals - both approach should work. When the modal component unmounts it should automatically close. Sorry we couldn't prioritize this fix earlier.

**Comment by mbaumbach** [comment]
@vividviolet Awesome, thanks for the fix!
